lib/combine_pdf/parser.rb

```diff
@@ -614,7 +614,7 @@ def catalog_pages(catalogs = nil, inheritance_hash = {})
             if catalogs[:Dests] # convert PDF 1.1 Dests to PDF 1.2+ Dests
               dests_arry = (@names_object[:Dests] ||= {})
               dests_arry = ((dests_arry[:referenced_object] || dests_arry)[:Names] ||= [])
-              ((catalogs[:Dests][:referenced_object] || catalogs[:Dests])[:referenced_object] || (catalogs[:Dests][:referenced_object] || catalogs[:Dests])).each {|k,v| next if CombinePDF::PDF::PRIVATE_HASH_KEYS.include?(k); dests_arry << unify_string(k.to_s); dests_arry << v; }
+              ((catalogs[:Dests][:referenced_object] || catalogs[:Dests])[:referenced_object] || (catalogs[:Dests][:referenced_object] || catalogs[:Dests])).each {|k,v| next if CombinePDF::PDF::PRIVATE_HASH_KEYS.include?(k); dests_arry << unify_string(+(k.to_s)); dests_arry << v; }
```

Comment:
I'm not a fan of the `+`, as it forces Ruby to duplicate the string and degrades performance.

Also, it seems a better approach might be to add a fix in `unify_string` rather than every time it's called.

I'll try to come up with a solution as well.